File: app/src/main/java/com/martinlaizg/geofind/utils/DateUtils.java
package com.martinlaizg.geofind.utils;

import java.sql.Date;
import java.util.Calendar;

public class DateUtils {

	public static final String DATE_FORMAT = "yyyy-MM-dd HH:mm:ss";
	public static final long TIME_TO_EXPIRE = 20 * 1000; // 20s

	public static boolean isDateExpire(Date fromDate) {
		return Calendar.getInstance().getTime().getTime() - fromDate.getTime() > TIME_TO_EXPIRE;
	}
}

File: app/src/main/java/com/martinlaizg/geofind/utils/DateUtils.kt
package com.martinlaizg.geofind.utils

import java.sql.Date
import java.util.*

object DateUtils {
    const val DATE_FORMAT = "yyyy-MM-dd HH:mm:ss"
    private const val TIME_TO_EXPIRE = 15 * 60 * 1000 // 15min * 60s * 1000ms
            .toLong()

    fun isDateExpire(fromDate: Date?): Boolean {
        return Calendar.getInstance().time.time - fromDate!!.time > TIME_TO_EXPIRE
    }
}